### DISCUSSION TITLE
Search Results Web Part Caching?

### DISCUSSION THREAD

**Discussion by poneilcsgp** [microsoft-search/pnp-modern-search#2057] [Ideas]
**Is your feature request related to a problem? Please describe.**
We use a number of search result web parts to expose search results for News Articles, Events, Birthdays, work Anniversaries etc.   These parts seem to be the bulk of the load time/contribute most of the perception of a "slow page load".   If we could enable caching and set cache expiration to say every 6, 12, or 24 hours, our page performance would likely improve significantly.

**Describe the solution you'd like**
Ability to Set a cache duration for the web part in which the part would simply fetch the previous results rather than hitting the search api.

**Describe alternatives you've considered**
I think I'd have to fork the existing project and build this myself/contribute back.  I'd love to do this but I think it would take me a (long) while to get up to speed enough to do this properly.

We do use LiveTiles on our main page so I'm also reaching out to them to see if they have any kind of caching ability in their parts that wrap modern web parts.

**Additional context**
Don't have any to share right now, but would be happy to add more context if you have questions or alternative suggestions.   Thanks for all your hard work, PnP has made a huge impact on SP development!

**Comment by wobba** [comment]
I think we potentially could use the PnPjs cache settings for this.

**Reply by poneilcsgp** [reply]
I see a few tags added to this.   I know you all have a ton of suggestions to sort through and lots of other updates, but I noticed the v4 and enhancement tag and was wondering if there were any plans for implementation?

**Comment by VerlindenEAM** [comment]
We also would like to have a caching feature, since we are using several results webparts and verticals.
